js/links.js

```diff
@@ -0,0 +1,242 @@
+ * and append it to the side block.
+ *
+ * @param {String} pageUrl - The URL of the translation to link to.
+ * @param {String} branchName - The name of the branch.
+ * @param {String} branchLang - The language code of the branch.
+ */
+function addTranslationLink(pageUrl, branchName, branchLang) {
+  var sideBlock = document.getElementsByClassName("side-block")[0];
+  var menuItems = Array.prototype.slice.call(
+    sideBlock.getElementsByClassName("menu-item")
+  );
+
+  // There is a translation, so unhide the side block if it is hidden
+  sideBlock.style.display = "";
+
+  // Create the new menu item
+  var newMenuItem = document.createElement("div");
+  newMenuItem.classList.add("menu-item");
+  // Record its branch's language code in the element
+  newMenuItem.setAttribute("name", branchLang);
+
+  // Create the bullet point image
+  var bullet = document.createElement("img");
+  bullet.setAttribute(
+    "src",
+    "//scp-wiki.wdfiles.com/local--files/nav:side/default.png"
+  );
+  bullet.setAttribute("alt", "default.png");
+  bullet.classList.add("image");
+  newMenuItem.appendChild(bullet);
+
+  // Create the actual link
+  var link = document.createElement("a");
+  link.setAttribute("href", pageUrl);
+  link.setAttribute("target", "_parent");
+  link.innerText = branchName;
+  newMenuItem.appendChild(link);
+
+  // Add the new menu item to the end of the side block by default
+  sideBlock.appendChild(newMenuItem);
+  // Then find the first existing menu item whose lang code is
+  // alphabetically greater than the new item, and move the new item to
+  // just before it
+  menuItems.some(function (menuItem) {
+    if (menuItem.getAttribute("name") > branchLang) {
+      sideBlock.insertBefore(newMenuItem, menuItem);
+      return true;
+    }
+  });
+}
+
+/**
+ * @callback findPagesCallback
+ * @param {String[]} fullnames
+ */
+
+/**
+ * In the given Wikidot site, searches for pages whose fullnames start with
+ * the given string.
+ *
+ * A 'fullname' is also referred to as a page's 'UNIX name'.
+ *
+ * @param {String} siteId - The numeric Wikidot site ID of the site to
+ * search.
+ * @param {String} fullname - The substring to compare fullnames against.
+ * If an underscore "_" is provided, all pages on the site will match.
+ * @param {findPagesCallback} callback - Will be called with the array of
+ * matching fullnames.
+ */
+function findPagesInSiteStartingWith(siteId, fullname, callback) {
+  var query = "&s=" + siteId + "&q=" + fullname;
+  var url = "/quickmodule.php?module=PageLookupQModule" + query;
+  var request = new XMLHttpRequest();
+  request.open("GET", url, true);
+  request.addEventListener("load", function () {
+    if (request.readyState === 4) {
+      var fullnames = [];
+      try {
+        if (request.status === 200) {
+          var response = JSON.parse(request.responseText);
+          // {"pages":[{"unix_name":"scp-xxx","title":"SCP-XXX"}]}
```

Comment:
I would include a "format:" or similar so it doesn't look like commented-out code

Reply:
f378899